appserver/admingui/common/src/main/java/org/glassfish/admingui/common/handlers/NewSSLHandlers.java

```diff
@@ -126,100 +129,85 @@ public static void convertCiphersItemsToStr(HandlerContext handlerCtx) {
-            eccCiphers = breakUpCiphers(eccCiphers, ciphers, "_ECDHE_"); //NOI18N
-            String[] ciphersList = (String[])eccCiphers.toArray(new String[eccCiphers.size()]);
-            return ciphersList;
-        }    
-        
-        private static String[] getEphemeralCiphers(Vector ciphers){
-            Vector ephmCiphers = breakUpCiphers(new Vector(), ciphers, "_DHE_RSA_"); //NOI18N
-            ephmCiphers = breakUpCiphers(ephmCiphers, ciphers, "_DHE_DSS_"); //NOI18N
-            String[] ciphersList = (String[])ephmCiphers.toArray(new String[ephmCiphers.size()]);
-            return ciphersList;
+        if(selectedCiphers != null){
+            for (String selectedCipher : selectedCiphers) {
+                sb.append(sep).append("+").append(selectedCipher);
+                sep = ",";
+            }
         }
-        
-        private static String[] getOtherCiphers(Vector ciphers){
-            Vector bitCiphers = filterCiphers(ciphers, BIT_CIPHERS);
-            String[] ciphersList = (String[])bitCiphers.toArray(new String[bitCiphers.size()]);
-            return ciphersList;
+        return sb.toString();
+    }
+
+    private static List<String> getCiphers(String[] allCiphers){
+        List<String> ciphers = new ArrayList<>();
+        if (allCiphers != null){
+            Collections.addAll(ciphers, allCiphers);
         }
-        
-        private static Vector filterCiphers(Vector ciphers, String[] filterList){
-            Vector listCiphers = new Vector();
-            if (ciphers != null){
-                for(int i=0; i<ciphers.size(); i++){
-                    String cipherName = ciphers.get(i).toString();
-                    if (Arrays.asList(filterList).contains(cipherName)){
-                        listCiphers.add(ciphers.get(i));
-                    }
-                }
-            }
-            return listCiphers;
+        return ciphers;
+    }
+
+    private static String[] getCommonCiphers(List<String> ciphers){
+        return filterCiphers(ciphers, COMMON_CIPHERS).toArray(new String[0]);
+    }
+
+    private static String[] getEccCiphers(List<String> ciphers){
+        List<String> eccCiphers = breakUpCiphers(new ArrayList<>(), ciphers, "_ECDH_"); //NOI18N
+        eccCiphers = breakUpCiphers(eccCiphers, ciphers, "_ECDHE_"); //NOI18N
+        return eccCiphers.toArray(new String[0]);
+    }
+
+    private static String[] getEphemeralCiphers(List<String> ciphers){
+        List<String> ephmCiphers = breakUpCiphers(new ArrayList<>(), ciphers, "_DHE_RSA_"); //NOI18N
+        ephmCiphers = breakUpCiphers(ephmCiphers, ciphers, "_DHE_DSS_"); //NOI18N
+        return ephmCiphers.toArray(new String[0]);
+    }
+
+    private static String[] getOtherCiphers(List<String> ciphers){
+        return filterCiphers(ciphers, BIT_CIPHERS).toArray(new String[0]);
+    }
+
+    private static List<String> filterCiphers(List<String> ciphers, Set<String> filterList){
```

Comment:
If `filterCiphers` would take a `Predicate<String>` evaluating if a cipher is accepted all the filters could be done with the same function (`breakUpCiphers` could also use filter) and in one pass (`breakUpCiphers` would not needed to be called multiple times) and the code would be more readable formulating the condition of the ciphers to keep as a lambda. WDYT?